partner_firstname/exceptions.py

```diff
@@ -0,0 +1,36 @@
+# -*- encoding: utf-8 -*-
+
+# Odoo, Open Source Management Solution
+# Copyright (C) 2014-2015  Grupo ESOC <www.grupoesoc.es>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from openerp import _, exceptions
+
+
+class PartnerNameError(exceptions.ValidationError):
+    def __init__(self, record, value=None):
+        self.record = record
+        self._value = value
+        self.name = _("Error(s) with partner %d's name.") % record.id
+
+    @property
+    def value(self):
+        raise NotImplementedError()
```

Comment:
It's a common pattern I follow for modules: do a base exception for that module and subclass it.  It helps with submodules and tests. However, seems like after finishing only one possible exception was left, so I will merge it as you say.